File: src/main/java/com/bottleh/studycodecollection/object/chap13/Rectangle.java
package com.bottleh.studycodecollection.object.chap13;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
public class Rectangle {

    private final int x;

    private final int y;

    @Getter
    @Setter
    private int width;

    @Getter
    @Setter
    private int height;

    public int getArea() {
        return width * height;
    }
}

File: src/main/java/com/bottleh/studycodecollection/kotlin/chap2/Rectangle.kt
package com.bottleh.studycodecollection.kotlin.chap2

class Rectangle(var height: Int, var width: Int) {
    val isSquare: Boolean
        get() {
            return height == width
        }
}

fun createUnitSquare(): Rectangle {
    return Rectangle(1, 1)
}